cips/cip-shwap_protocol.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Shwap Protocol 
+description: Shwap - the new extensible sampling protocol 
+author: Hlib Kanunnikov (@Wondertan)
+discussions-to: https://forum.celestia.org/t/cip-shwap-protocol/1551
+status: Draft
+type: Standards Track
+category: Data Availability, Networking
+created: 2024-02-02
+---
+
+## Abstract
+
+This document describes a high level overview of the Shwap p2p protocol. Shwap provides scalable and extensible 
```

Comment:
What is the rationale for the name "Shwap"?

Reply:
There is no "rationale". But it is short for SharesSwap and is something that is to remember, short and sounding. It also gives me the vibe of being fast, like a snap.

Reply:
Thanks! We could add something like this to the CIP to front run the same question by other readers:

> This document describes a high level overview of the Shwap p2p protocol. Shwap is a portmanteau of share and swap.

Reply:
Good suggestion!